sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -80,42 +80,37 @@ def _generate_recommendations(self):
         )
         found_dict = {l["product_id"][0]: l for l in found_lines}
         RecomendationLine = self.env["sale.order.recommendation.line"]
+        wiz_line_ids = self.env["sale.order.recommendation.line"]
         existing_product_ids = []
         # Add products from sale order lines
         for line in self.order_id.order_line:
             found_line = found_dict[line.product_id.id]
-            new_line = RecomendationLine.new({
+            new_line = RecomendationLine.create({
```

Comment:
In general terms, creating records in an onchange is not a very good idea... This is a transient model, so records get removed every now and then, so maybe we can make an exception?

:thinking: Have you tried converting each `.new()` to `(0, 0, {...})`? I think that behind the scenes both get casted to the same operation, but @rco-odoo told me recently that this wasn't 100% the case. At least it's worth trying?